src/main/java/software/amazon/nio/spi/s3/util/StringUtils.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.nio.spi.s3.util;
+
+import software.amazon.nio.spi.s3.S3Path;
+
+/**
+ *
```

Comment:
Needs javadoc, including what happens to elements before the `startAt()`.

A more informative name might be useful as this is not just a normal `String.join(separator elements)`